plugin/virustotal/vtreport.py

```diff
@@ -0,0 +1,404 @@
+
+    ip_report = self.contacted_ips[ip]
+    ip_details = []
+    ip_analysis = ip_report.get('last_analysis_stats')
+    total = ip_analysis['harmless'] + ip_analysis['malicious']
+    total += ip_analysis['suspicious'] + ip_analysis['undetected']
+    ip_analysis = str(ip_analysis['malicious']) + '/' + str(total)
+    ip_details.append(ip)
+    ip_details.append(ip_analysis)
+    ip_details.append(ip_report.get('country'))
+    return ip_details
+
+  def get_embedded_ips(self):
+    return self.embedded_ips.keys()
+
+  def get_embedded_ip(self, ip):
+    """Return a list containing IPs + detections + country."""
+
+    ip_report = self.embedded_ips[ip]
+    ip_details = []
+    ip_analysis = ip_report.get('last_analysis_stats')
+    total = ip_analysis['harmless'] + ip_analysis['malicious']
+    total += ip_analysis['suspicious'] + ip_analysis['undetected']
+    ip_analysis = str(ip_analysis['malicious']) + '/' + str(total)
+    ip_details.append(ip)
+    ip_details.append(ip_analysis)
+    ip_details.append(ip_report.get('country'))
+    return ip_details
+
+  def get_contacted_urls(self):
+    return self.contacted_urls.keys()
+
+  def get_contacted_url(self, url):
+    """Return a list containing URL + detections."""
+
+    url_report = self.contacted_urls[url]
+    url_details = []
+    url_analysis = url_report.get('last_analysis_stats')
+    total = url_analysis['harmless'] + url_analysis['malicious']
+    total += url_analysis['suspicious'] + url_analysis['undetected']
+    url_analysis = str(url_analysis['malicious']) + '/' + str(total)
+    url_details.append(url)
+    url_details.append(url_analysis)
+    url_details.append(url_report.get('last_analysis_date'))
+    return url_details
+
+  def get_embedded_urls(self):
+    return self.embedded_urls.keys()
+
+  def get_embedded_url(self, url):
+    """Return a list containing URL + detections + scanned time."""
+
+    url_report = self.embedded_urls[url]
+    url_details = []
+    url_details.append(url)
+    detections = url_report.get('last_analysis_stats')
+
+    if detections:
+      total = detections['harmless'] + detections['malicious']
+      total += detections['suspicious'] + detections['undetected']
+      ratio = str(detections['malicious']) + '/' + str(total)
+    else:
+      ratio = None
+
+    url_details.append(ratio)
+    url_details.append(url_report.get('last_analysis_date'))
+    return url_details
+
+  def get_contacted_domains(self):
+    return self.contacted_domains.keys()
+
+  def get_contacted_domain(self, domain):
+    """Return a list containing IP + detections."""
+
+    domain_report = self.contacted_domains[domain]
+    domain_details = []
+    domain_analysis = domain_report.get('last_analysis_stats')
+    total = domain_analysis['harmless'] + domain_analysis['malicious']
+    total += domain_analysis['suspicious'] + domain_analysis['undetected']
+    domain_analysis = str(domain_analysis['malicious']) + '/' + str(total)
```

Comment:
These 3 lines repeat a lot, maybe extract to a common function?

def _ratio(o, a, b, c, d):
  num = o[a] + o[b]
  denom = o[c] + o[d]
  return '%d/%d' % (num, denom)


... altho many if not all cases use the same key names, so maybe better:

@staticmethod
def _ratio(o):
  num = o['harmless'] + o['malicious']
  denom = o['suspicious'] + o['undetected']
  return '%d/%d' % (num, denom)